src/sentry/utils/types.py

```diff
@@ -1,29 +1,34 @@
+from __future__ import annotations
+
 import typing
 
+from typing_extensions import TypeGuard
 from yaml.parser import ParserError
 from yaml.scanner import ScannerError
 
 from sentry.utils.yaml import safe_load
 
 __all__ = ("InvalidTypeError", "Any", "Bool", "Int", "Float", "String", "Dict", "Sequence")
 
+T = typing.TypeVar("T")
```

Comment:
Is there a specific reason we import entire `typing` module rather than importing what we use?

Reply:
this module defines a class called `Any` so it would collide